editor/animation_track_editor.cpp

```diff
@@ -7319,27 +7319,26 @@ void AnimationTrackEditor::_update_snap_unit() {
 	}
 
 	if (timeline->is_using_fps()) {
-		snap_unit = 1.0 / step->get_value();
+		snap_unit = Math::snapped(1.0 / step->get_value(), FPS_STEP_FRACTION);
 	} else {
-		double integer;
-		double fraction = Math::modf(step->get_value(), &integer);
-		fraction = 1.0 / Math::round(1.0 / fraction);
-		snap_unit = integer + fraction;
+		snap_unit = Math::snapped(step->get_value(), SECOND_DECIMAL);
```

Comment:
Will a snap of 0.333 accumulate correctly?
I remember that the modf was needed so that when 0.333 was accumulated 3 times, it would snap to 1.0 instead of 0.999.